### PR TITLE
Partially reverting Divider + bringing back Separator

### PR DESCRIPTION
**Pull request by sophialee0416**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Binary change:

| File | Before | After | Delta |
|------|--------|-------|-------|
| libFluentUI.a | 28,947,552 bytes | 28,723,448 bytes | -224,104 bytes |

Divider was brought to main in #1343, but due to the large binary increase, we unfortunately have to revert this change and bring Separator back. This has already been done in 0.9 (this PR contains a cherry-pick from #1369) but this time we'll leave the Divider files in the framework and exclude them in the list of Compile Sources for the FluentUILib target.

### Verification

Build + run test app, app is back to using old separator and there are no use cases of Divider.
| Before | After | 
|------|--------|
| image | image | 
### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1441)

### CODE REVIEW

**Comment by edjamesmsft** [comment]
Could you include before and after screen shots?

File: ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -72,15 +72,15 @@
...
-		EC6A71EC273DE6DF0076A586 /* DividerDemoController.swift in Sources */ = {isa = PBXBuildFile; fileRef = EC6A71EB273DE6DF0076A586 /* DividerDemoController.swift */; };
+		EC24DBC628B97EB70026EF92 /* PopupMenuObjCDemoController.m in Sources */ = {isa = PBXBuildFile; fileRef = EC24DBC528B97EB70026EF92 /* PopupMenuObjCDemoController.m */; };
 		FC414E3725888BC300069E73 /* CommandBarDemoController.swift in Sources */ = {isa = PBXBuildFile; fileRef = FC414E3625888BC300069E73 /* CommandBarDemoController.swift */; };
 		FDCF7C8321BF35680058E9E6 /* SegmentedControlDemoController.swift in Sources */ = {isa = PBXBuildFile; fileRef = FDCF7C8221BF35680058E9E6 /* SegmentedControlDemoController.swift */; };
 /* End PBXBuildFile section */
 
 /* Begin PBXFileReference section */
+		0A0516BF2940A407006DE0A2 /* DividerDemoController_SwiftUI.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = DividerDemoController_SwiftUI.swift; sourceTree = "<group>"; };
+		0A0516C02940A407006DE0A2 /* DividerDemoController.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = DividerDemoController.swift; sourceTree = "<group>"; };
```
[the remark this answers is not in the document]
Comment (sophialee0416) [reply ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/project.pbxproj:83]: The demo doesn't build. If I want it to build then I need to add all the files to the target membership which will then add it back to the library which I don't think we want.

[the remark this answers is not in the document]
Comment (huwilkes) [reply ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/project.pbxproj:83]: Is the divider demo also not included in a target, then? So we're just keeping it around to preserve the code?

File: MicrosoftFluentUI.podspec [hunk 2]
```diff
@@ -135,12 +135,6 @@ fi', :execution_position => :before_compile }
                              "ios/FluentUI/Extensions/**/*.{swift,h}"]
   end
 
-  s.subspec 'Divider_ios' do |divider_ios|
-    divider_ios.platform = :ios
-    divider_ios.dependency 'MicrosoftFluentUI/Core_ios'
-    divider_ios.source_files = ["ios/FluentUI/Divider/**/*.{swift,h}"]
-  end
```
Comment (harrieshin) [inline MicrosoftFluentUI.podspec:142]: assuming that htis PR didn't deelete the divder folder, that means the cocoapod consuming as framework will have both separator and dvider in the same library. in that case the subspec should still stay the same?

File: ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/project.pbxproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (sophialee0416) [reply ios/FluentUI.Demo/FluentUI.Demo.xcodeproj/project.pbxproj:83]: Resolved offline, but yes we are keeping Divider around to preserve the code and eventually see if we can replace it with Separator in the near future.

**Review by huwilkes — APPROVED** [review]

**Review by harrieshin — APPROVED** [review]

**Pull request merged**